src/watts_ui/watts_ui.py

```diff
@@ -0,0 +1,610 @@
+        app_plugin = watts.PluginSAS(
+            template_file=plugin['template'],
+            executable=exec,
+            extra_inputs=plugin['extra_inputs'],
+            extra_template_inputs=plugin['extra_template_inputs'],
+            show_stdout=plugin['show_stdout'],
+            show_stderr=plugin['show_stderr'])
+
+        if 'conv_channel' in plugin:
+            app_plugin.conv_channel = Path(plugin['conv_channel'])
+        if 'conv_primar4' in plugin:
+            app_plugin.conv_primar4 = Path(plugin['conv_primar4'])
+
+    elif plugin['code'].upper() == 'SERPENT':
+        app_plugin = watts.PluginSerpent(
+            template_file=plugin['template'],
+            executable=exec,
+            extra_inputs=plugin['extra_inputs'],
+            extra_template_inputs=plugin['extra_template_inputs'],
+            show_stdout=plugin['show_stdout'],
+            show_stderr=plugin['show_stderr'])
+
+    elif plugin['code'].upper() == 'DAKOTA':
+        app_plugin = watts.PluginDakota(
+            template_file=plugin['template'],
+            executable=exec,
+            extra_inputs=plugin['extra_inputs'],
+            extra_template_inputs=plugin['extra_template_inputs'],
+            auto_link_files=plugin['auto_link_files'],
+            show_stdout=plugin['show_stdout'],
+            show_stderr=plugin['show_stderr'])
+
+    elif plugin['code'].upper() == 'ABCE':
+        app_plugin = watts.PluginABCE(
+            template_file=plugin['template'],
+            executable=exec,
+            extra_inputs=plugin['extra_inputs'],
+            extra_template_inputs=plugin['extra_template_inputs'],
+            show_stdout=plugin['show_stdout'],
+            show_stderr=plugin['show_stderr'])
+
+    elif plugin['code'].upper() == 'MCNP':
+        app_plugin = watts.PluginMCNP(
+            template_file=plugin['template'],
+            executable=exec,
+            extra_inputs=plugin['extra_inputs'],
+            extra_template_inputs=plugin['extra_template_inputs'],
+            show_stdout=plugin['show_stdout'],
+            show_stderr=plugin['show_stderr'])
+
+    elif plugin['code'].upper() == 'OPENMC':
+        sys.path.insert(0, os.getcwd())
+        from openmc_template import build_openmc_model
+
+        app_plugin = watts.PluginOpenMC(
+            model_builder=build_openmc_model,
+            extra_inputs=plugin['extra_inputs'],
+            show_stdout=plugin['show_stdout'],
+            show_stderr=plugin['show_stderr'])  # show only error
+
+    else:
+        raise RuntimeError("Please provide the correct application name.")
+
+    # Set 'extra_args' if available
+    if 'extra_args' in plugin:
+        app_plugin(watts_params, extra_args=plugin['extra_args'])
+
+    # Run plugins and save to app_result
+    app_result = app_plugin(watts_params)
+
+    # Store output data from app_result to watts_params
+    # Special treatment for OpenMC
+    if plugin['code'].upper() == 'OPENMC':
```

Comment:
could this be moved above to have a 'post-run' logic added in plugin specification? I will need to do the same for PyARC.

Reply:
We decided to disregard this in the last meeting.